### ISSUE TITLE
pystac `1.11.0` requires Python `>=3.10`

### ISSUE THREAD

**Issue by jtherrmann** [ASFHyP3/asf-stac#651]
**Jira:** https://asfdaac.atlassian.net/browse/TOOL-3216

*Note: The above link is accessible only to members of ASF.*

----------------------------------------------------------------------------------------------------

https://github.com/ASFHyP3/asf-stac/pull/644 is blocked because pystac `1.11.0` requires Python `>=3.10`, but `asf-stac` is on `3.9`: https://github.com/ASFHyP3/asf-stac/blob/develop/environment.yml

We should upgrade Python, preferably to the latest release. Blocked by https://github.com/ASFHyP3/.github-private/issues/26

**Comment by jtherrmann** [comment]
Update on the status of https://github.com/ASFHyP3/asf-stac/pull/644:
  
I merged `develop` into the dependabot branch on Oct 9, 2024, which caused dependabot to comment on Jan 27, 2025:
  
>A newer version of pystac exists, but since this PR has been edited by someone other than Dependabot I haven't updated it. You'll get a PR for the updated version as normal once this PR is merged.

I then closed the PR on Feb 11, 2025 and dependabot replied:

>OK, I won't notify you again about this release, but will get in touch when a new version is available.

However, since then, [pystac v1.12.2](https://github.com/stac-utils/pystac/releases/tag/v1.12.2) released on Feb 21, 2025, so I would have expected to see a dependabot PR open on Monday Feb 24, per [dependabot.yml](https://github.com/ASFHyP3/asf-stac/blob/develop/.github/dependabot.yml), but that didn't happen.

So, I re-opened the PR on Feb 26, 2025, and it can sit there until we upgrade the Python version, at which point we should merge it and then confirm that dependabot continues bumping pystac versions as expected.

I also ran `@dependabot recreate` in the hope that it would re-create the PR from scratch and then we'd see it superseded by a PR for the latest version, but it failed with:

>Dependabot does not support your Python version. Because of this, Dependabot cannot update this pull request.

which I don't understand, since this repo should be using Python 3.9 everywhere and I believe the last Python version dropped by dependabot [should be 3.8](https://github.blog/changelog/2025-01-06-closing-down-notice-dependabot-will-no-longer-support-python-version-3-8/).

Then there's https://github.com/ASFHyP3/asf-stac/pull/675 which is failing with:
  
```
ERROR: Cannot install -r /home/runner/work/asf-stac/asf-stac/requirements-apps-api.txt (line 2) and pypgstac==0.9.2 because these package versions have conflicting dependencies.
```

`requirements-apps-api.txt` line 2 is `stac-fastapi.pgstac==3.0.1`, though I'm not exactly sure what the dependency conflict is (we may have encountered this issue with previous versions of these packages, I don't remember).

Some notes I took when this PR first opened indicate that the conflict is `stac-fastapi-pgstac 3.0.1 depends on pypgstac==0.8.*` but I haven't yet found that message in the log for the latest failed pytest workflow (possibly building locally would help).

Note that I also merged `develop` into this branch (same as the other PR above), so I assume dependabot will ignore newer versions until we merge this PR, at which point we should confirm that dependabot resumes bumping versions of this package.

**Comment by jtherrmann** [comment]
This issue is no longer needed because dependabot should open a PR to bump pystac upon upgrading the Python version. See https://github.com/ASFHyP3/asf-stac/pull/644#issuecomment-2787803069
